cookbook/security/api_key_authentication.rst

```diff
@@ -318,6 +318,10 @@ using the ``simple_preauth`` key:
                     simple_preauth:
                         authenticator: apikey_authenticator
 
+            providers:
+                simple_preauth:
+                    id: your_api_key_user_provider
+
```

Comment:
The class behind this service would be the `ApiKeyUserProvider` class that we create in this article, right? If so, I'd like your thought on going even a bit further:

1) Change the name from `simple_preauth` to something else, because I don't think this needs to correspond with the authentication name under the firewall (so using something else will make that obvious)

2) Change the key used here to be something like `api_key_user_provider`

3) After we show them the `ApiKeyUserProvider`, we just kind of link to the "custom user provider" article. What do you think about adding the service configuration below that code block and also show the `security.yml` configuration? I don't want to duplicate, but I feel like *maybe* we're leaving the user hanging without enough details.

Thanks :)